### ISSUE TITLE
How to use with storyboard tabbed app

### ISSUE THREAD

**Issue by alexrainman** [Baseflow/Xamarin-Sidebar#13]
Is it possible?

**Comment by pilucdes** [comment]
How it work with storyboard :
1. All view can be done in storyboard but you can't add segue from storyboard since the navigation controller is not in storyboard but in code behind
1. If you want to push you will have to instance your view from the code "Storyboard.InstantiateViewController" and push manually from the navigation controller that you have make an instance in the root view and hold reference in appdelegate(well this is how i do it)

This is where all start, create a first view that will be your root in the storyboard. Set it as rootview (starting point) in the storyboard. This view is empty since it's the container

Here is my rootview  ViewDidLoad function :

```
        //Set Navigation controller reference so we can use it everywhere
        AppDelegate appD = UIApplication.SharedApplication.Delegate as AppDelegate;
        appD.GETSET_NavController = new UINavigationController();


        VCConnection vConnect = (VCConnection)Storyboard.InstantiateViewController("VCConnection");
        VCLeftMenu vLeftMenu = (VCLeftMenu)Storyboard.InstantiateViewController("VCLeftMenu");
        appD.GETSET_NavController.PushViewController(vConnect,false);
        appD.GETSET_SidebarController = new SidebarController(this, appD.GETSET_NavController, vLeftMenu);
        appD.GETSET_SidebarController.MenuWidth = 250;
        appD.GETSET_SidebarController.ReopenOnRotate = false;
        appD.GETSET_SidebarController.IsOpening += appD.GETSET_Menu.UpdatePicturesToSynch;
```

VConnection is my default view to be shown in the content which is also created in storyboard.
VCLeftMenu is the view that will be shown in the  sidebarMenuNavController which is also created in storyboard.
appD.GETSET_NavController is the nav controller created to be able to push manually in the code.

I also hold a reference to the sidebarController in case i need it some where...

Basically, in this case, the storyboard utility is to only create view using a UI and apply constraint to it.
Just don't use segue in the storyboard because u will run into instance error,

When i want to change the content view i simply do this :

```
      UIViewController vcSupport = (UIViewController)Storyboard.InstantiateViewController("VCSupport");
        GET_APPD.GETSET_NavController.PushViewController(vcSupport,true);
```

I instance from storyboard then i push with the reference navigation controller in the appdelegate

**Comment by alexrainman** [comment]
Thanks for the quick response. Have it working with .xib file and tab bar control anyway i will give it a try!

**Comment by alexrainman** [comment]
Cant make it work. I get Object reference not set to an instance of an object when i try this line. VCConnection vConnect = (VCConnection)Storyboard.InstantiateViewController("VCConnection");
Can you provide working example?

**Comment by pilucdes** [comment]
What is null exactly ?? can you debug it ?  did you do a view in the storyboard named VCConnection ? when u create a view in storyboard don't forget to link a name (Class in custom class) to it so it will create a csharp class (.cs).

**Comment by alexrainman** [comment]
I worked with Storyboards before. Each view has ID and controller class associated. I get that error when i try yo instance the content view from storyboard.

**Comment by pilucdes** [comment]
WIll upload a basic example in a minute

**Comment by alexrainman** [comment]
You rock! Thanks.

**Comment by pilucdes** [comment]
Here is a working solution : https://drive.google.com/file/d/0B8T5MI2VT7ksVFg4YWdpanF5OTg/view?usp=sharing

**Comment by alexrainman** [comment]
It works. My problem was on the AppDelegate. Thanks a lot!

**Comment by whitneyland** [comment]
Is it allowed to move the NavigationController into the StoryBoard as well?  This would allow using the designer to set styles and properties for the Nav as well.

**Comment by pilucdes** [comment]
You can set styles and properties in the code behind too.
You can still try to affect the Navigation controller from the Storyboard in the Sidebar and see if it's works

**Comment by whitneyland** [comment]
Yes, you do can do everything in code and don't need any storyboards at all. :).  However, for many scenarios the more that can be in a storyboard, the better.

So I'm wondering, could the navigation controller from your example be put there for more simplicity in the code.

If not no problem, I appreciate your contributions and code.

**Comment by pilucdes** [comment]
I remember running into some instancing problem if i use the navigation controller from the storyboard. That's why i decide to create it from code instead.

Like i said, you can still try to affect the Navigation controller from the Storyboard in the Sidebar constructor and see if it's works when using the segue from the storyboard

**Comment by jdehlin** [comment]
Thanks for helping with this issue @pilouk!
